src/compoents/BookMentorForm.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { useEffect, useState } from 'react';
+import {Card,
+    CardMedia,
+    CardContent,
+    Typography,
+    Button,
+    Chip,
+    Box,
+    Rating,
+    Grid,
+    FormControl,
+    FormLabel,
+    FormHelperText,
+    Autocomplete,
+    TextField,
+    TextareaAutosize,} from '@mui/material';
+
+import TipTapEditor from './TipTapEditor';
+import { useAuth } from '../providers/AuthProvider';
+import { useStatus } from '../providers/MsgStatusProvider';
+import { createFilterOptions } from '@mui/material/Autocomplete';
+
+const filter = createFilterOptions();
+
+function BookMentorForm({ talentDetails, setOpen }) {
+    const [ formData, setFormData ] = useState({});
+    const [ supportAreas, setSupportAreas ] = useState();
+    const { token } = useAuth();
+    const { setStatusMessage, setIsAlertOpen, setStatusType } = useStatus();
+
+    useEffect(() => {
+        // This should be replaced with the appropriate API call
+        const url = process.env.REACT_APP_API_BASE_URL + 'mentorship/details/?fields=mentor_support_areas';
+        fetch(url)
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then(data => {
+                setSupportAreas(data.mentor_support_areas);
+            })
+            .catch(error => {
+                console.error('Fetch error:', error);
+            });
+    }, []);
+
+    const handleSupportAreasChange = (event, newValue) => {
+        console.log(formData, 'formData');
+        // Call the function passed from the parent component to update the formData state
+        setFormData(preValue => ({
+            ...preValue,
+            mentor_support_areas: newValue.map(item => item.id),
+        }));
+    };
+    const handleSubmit = () => {
+        console.log(talentDetails);
+        const url = process.env.REACT_APP_API_BASE_URL + `mentorship/mentor/${talentDetails.user.id}/connect/roster/add`;
+        fetch(url, {
+            method: 'POST',
+            headers: {
+                Authorization: `Token ${token}`,
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(formData),
+        })
+            .then(response => {
+                if (!response.ok) {
+                    setStatusMessage('Sorry, we ran into an issue, please try again');
+                    setIsAlertOpen(true);
+                    setStatusType('error');
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            })
+            .then(data => {
+                setOpen(false);
+                setStatusMessage('Saved your review');
```

Comment:
Can we implement the hook in `src/hooks/useStatusMessage.js` as in all other places in the app? it helps keep the code clean and happy :smiley: